### PR TITLE
CGMES export to version 3

### PR DESCRIPTION
**Pull request by zamarrenolm**
**Please check if the PR fulfills these requirements** *(please use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes)*
- [X] The commit message follows our guidelines
- [X] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem ?** *If so, link to this issue using `'#XXX'` and skip the rest*

**What kind of change does this PR introduce?** *(Bug fix, feature, docs update, ...)*
Ability to export go CGMES 3. It can be requested by the user setting export param `iidm.export.cgmes.cim-version` to 100.

**What is the current behavior?** *(You can also link to an open issue here)*

**What is the new behavior (if this is a feature change)?**

**Does this PR introduce a breaking change or deprecate an API?** *If yes, check the following:*
- [ ] The *Breaking Change* or *Deprecated* label has been added
- [ ] The migration guide has been updated in the github wiki *(What changes might users need to make in their application due to this PR?)*

**Other information**:

(if any of the questions/checkboxes don't apply, please delete them entirely)

### CODE REVIEW

File: cgmes/cgmes-model/src/main/java/com/powsybl/cgmes/model/CgmesNamespace.java [hunk 1]
```diff
@@ -21,32 +22,70 @@ private CgmesNamespace() {
...
+        // FIXME(Luma) This is legacy behaviour, we do not consider CIM14 valid in this check
+        // But I think we do not need to support 14 separately?
+        return namespaces.contains(CIM_16_NAMESPACE) || namespaces.contains(CIM_100_NAMESPACE);
+    }
+
+    public static boolean isValidAny14(Set<String> namespaces) {
+        return namespaces.contains(CIM_14_NAMESPACE);
+    }
+
+    public static String getCim(int cimVersion) {
```
Comment (miovd) [inline cgmes/cgmes-model/src/main/java/com/powsybl/cgmes/model/CgmesNamespace.java:75]: Why did you rename this method?

File: cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/export/elements/ControlAreaEq.java [hunk 2]
```diff
@@ -30,7 +30,7 @@ public static void write(String id, String controlAreaName, String energyIdentif
         writer.writeCharacters(energyIdentificationCodeEIC);
         writer.writeEndElement();
         writer.writeEmptyElement(cimNamespace, "ControlArea.type");
-        writer.writeAttribute(RDF_NAMESPACE, CgmesNames.RESOURCE, CONTROL_AREA_TYPE);
+        writer.writeAttribute(RDF_NAMESPACE, CgmesNames.RESOURCE, cimNamespace + CONTROL_AREA_TYPE);
```
Comment (miovd) [inline cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/export/elements/ControlAreaEq.java:33]: Isn't there a '#' missing here?

File: cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/export/elements/RegulatingControlEq.java [hunk 3]
```diff
@@ -29,7 +29,7 @@ public static void write(String id, String regulatingControlName, String termina
         writer.writeEmptyElement(cimNamespace, "RegulatingControl.Terminal");
         writer.writeAttribute(RDF_NAMESPACE, CgmesNames.RESOURCE, "#" + terminalId);
         writer.writeEmptyElement(cimNamespace, "RegulatingControl.mode");
-        writer.writeAttribute(RDF_NAMESPACE, CgmesNames.RESOURCE, REGULATING_CONTROL_VOLTAGE);
+        writer.writeAttribute(RDF_NAMESPACE, CgmesNames.RESOURCE, cimNamespace + REGULATING_CONTROL_VOLTAGE);
```
Comment (miovd) [inline cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/export/elements/RegulatingControlEq.java:32]: Same remark

File: cgmes/cgmes-model/src/main/java/com/powsybl/cgmes/model/CgmesNamespace.java [hunk 4]
```diff
@@ -21,32 +22,70 @@ private CgmesNamespace() {
...
+
+    public static boolean isValidAny(Set<String> namespaces) {
+        // FIXME(Luma) This is legacy behaviour, we do not consider CIM14 valid in this check
+        // But I think we do not need to support 14 separately?
+        return namespaces.contains(CIM_16_NAMESPACE) || namespaces.contains(CIM_100_NAMESPACE);
+    }
+
+    public static boolean isValidAny14(Set<String> namespaces) {
+        return namespaces.contains(CIM_14_NAMESPACE);
+    }
```
Comment (miovd) [inline cgmes/cgmes-model/src/main/java/com/powsybl/cgmes/model/CgmesNamespace.java:73]: The methods names are not really explicit.... What did you mean?

**Review by miovd — CHANGES_REQUESTED** [review]

**Comment by miovd** [comment]
@zamarrenolm Why is missing for this PR to be ready to review?

File: cgmes/cgmes-model/src/main/java/com/powsybl/cgmes/model/CgmesOnDataSource.java [hunk 5]
```diff
@@ -75,9 +76,7 @@ public String baseName() {
     }
 
     private static boolean isCimNamespace(String ns) {
-        // Until CIM16 the CIM namespace contained the string "CIM-schema-cim<versionNumber>#"
-        // Since CIM100 the namespace seems to follow the pattern "/CIM<versionNumber>#"
-        return CIM_NAMESPACES.contains(ns) || CIM_100_PLUS_NAMESPACE_PATTERN.matcher(ns).matches();
+        return CgmesNamespace.isValid(ns);
```
Comment (miovd) [inline cgmes/cgmes-model/src/main/java/com/powsybl/cgmes/model/CgmesOnDataSource.java:79]: You don't need a method for this

File: cgmes/cgmes-model/src/main/java/com/powsybl/cgmes/model/CgmesNamespace.java (hunk 1 above)
Comment (zamarrenolm) [reply cgmes/cgmes-model/src/main/java/com/powsybl/cgmes/model/CgmesNamespace.java:65]: It seemed redundant. It is a static method, and the class name already talks about namespace, so the calls will look like: `CgmesNamespace.getCim(cimVersion);`

File: cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/export/elements/ControlAreaEq.java (hunk 2 above)
Comment (zamarrenolm) [reply cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/export/elements/ControlAreaEq.java:33]: All namespaces finish with a dash character, `#`, so I think it is ok.

File: cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/export/elements/RegulatingControlEq.java (hunk 3 above)
Comment (zamarrenolm) [reply cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/export/elements/RegulatingControlEq.java:32]: It is ok for CIM namespaces

**Comment by zamarrenolm** [comment]
> @zamarrenolm What is missing for this PR to be ready to review?

There are some attributes that have been renamed (related to generators). Also, `mRID` is always present in the sample XML files, in addition to `rdf:ID`, I have to double check if it is required or optional (I think it is required).

**Comment by zamarrenolm** [comment]
The unit test `ExportToCimVersionTest` contains now a basic verification that we can export to CGMES 3 and reimport, and we obtain a Network with the same number of elements of the original one. 

This could work as a first beta-version of the export to CGMES 3.

Known issues: 
- `GeneratingUnit.initialP` should not be exported.
- `OperationalLimit` `TypeName` should be exported in a different way (`eu` extension instead of `entsoe`).
- `OperationalLimit` value should be exported as `normalValue`.
- Every object should have mRID exported, in addition to `rdf:ID` (the `mRID` does not require a `_` prefix).

**Review by miovd — APPROVED** [review]
